### PR TITLE
Mistral workflow with text models

### PR DESCRIPTION
**Pull request by SurajBaloni**
Guide for Mistral workflow with text models

### CODE REVIEW

File: guide/14-deep-learning/mistral_workflow_with_text_models.ipynb [hunk 1]
```diff
@@ -0,0 +1,617 @@
+{
```
Comment (r-deo) [inline guide/14-deep-learning/mistral_workflow_with_text_models.ipynb:1]: <p>Specify the minimum number of examples to 6</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1794/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='39'/>

**Comment by SurajBaloni** [comment]
@BP-Ent please review this guide.

File: guide/14-deep-learning/mistral_workflow_with_text_models.ipynb (hunk 1 above)
Comment (SurajBaloni) [reply guide/14-deep-learning/mistral_workflow_with_text_models.ipynb:1]: done

File: guide/14-deep-learning/mistral_workflow_with_text_models.ipynb [hunk 2]
```diff
@@ -0,0 +1,621 @@
...
```
Comment (kapil-varshney) [inline guide/14-deep-learning/mistral_workflow_with_text_models.ipynb:1]: <p>Use Mistral LLM for Text Classification and Entity Recognition</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1794/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

Comment (kapil-varshney) [inline guide/14-deep-learning/mistral_workflow_with_text_models.ipynb:1]: <p>The introduction needs improvement.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1794/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

Comment (kapil-varshney) [inline guide/14-deep-learning/mistral_workflow_with_text_models.ipynb:1]: <p>Reword the first line</p><p> </p><p>*with arcgis.learn</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1794/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

Comment (kapil-varshney) [inline guide/14-deep-learning/mistral_workflow_with_text_models.ipynb:1]: <p>Once the data is prepared, <code>TextClassifier</code> model object can be instantiated as below with the following parameters:</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1794/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='13'/>

Comment (kapil-varshney) [inline guide/14-deep-learning/mistral_workflow_with_text_models.ipynb:1]: <p>Add a few words about not using databunch or data=None</p><p>*only a few examples</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1794/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='17'/>

Comment (kapil-varshney) [inline guide/14-deep-learning/mistral_workflow_with_text_models.ipynb:1]: <p>"zip" not to be mentioned.</p><p>This method saves the model weights and creates a Deep Learning Package (.dlpk).</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1794/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='27'/>

File: guide/14-deep-learning/use_mistral_llm_for_text_classification_and_entity_recognition.ipynb [hunk 3]
```diff
@@ -0,0 +1,617 @@
+{
```
Comment (kapil-varshney) [inline guide/14-deep-learning/use_mistral_llm_for_text_classification_and_entity_recognition.ipynb:1]: <p>*Employing attention mechanisms...</p><p> </p><p>Second and Third para can be one para.</p><p> </p><p>Last line:</p><p>Such design features in its architecture equip Mistral 7B for exceptional performance, particularly in tasks related to language comprehension and generation. In this guide we see how we can use the Mistral LLM for text classification and named entity recognition.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1794/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

Comment (kapil-varshney) [inline guide/14-deep-learning/use_mistral_llm_for_text_classification_and_entity_recognition.ipynb:1]: <p>Follow these steps to download and install the Mistral model backbone:</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1794/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

Comment (kapil-varshney) [inline guide/14-deep-learning/use_mistral_llm_for_text_classification_and_entity_recognition.ipynb:1]: <p>prompt: [Optional] Text string describing the task and its guardrails.</p><p> </p><p>Make similar edits in the next section.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1794/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='14'/>

Comment (kapil-varshney) [inline guide/14-deep-learning/use_mistral_llm_for_text_classification_and_entity_recognition.ipynb:1]: <p>You can create a <code>TextClassifier</code> model with mistral backbone without a large dataset and by using only a few examples.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1794/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='17'/>

Comment (kapil-varshney) [inline guide/14-deep-learning/use_mistral_llm_for_text_classification_and_entity_recognition.ipynb:1]: <p><code>examples</code>: User defined examples to the mistral model, in python dictionary format.</p><p> </p><p>prompt: same as earlier section</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1794/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='18'/>

Comment (kapil-varshney) [inline guide/14-deep-learning/use_mistral_llm_for_text_classification_and_entity_recognition.ipynb:1]: <p>To classify text using the mistral model...</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1794/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='21'/>

Comment (kapil-varshney) [inline guide/14-deep-learning/use_mistral_llm_for_text_classification_and_entity_recognition.ipynb:1]: <p>To load a mistral model saved earlier use <code>from_model</code> method from <code>TextClassifier</code> class.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1794/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='24'/>

Comment (kapil-varshney) [inline guide/14-deep-learning/use_mistral_llm_for_text_classification_and_entity_recognition.ipynb:1]: <p>path to emd or dlpk?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1794/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='25'/>

Comment (kapil-varshney) [inline guide/14-deep-learning/use_mistral_llm_for_text_classification_and_entity_recognition.ipynb:1]: <p>Same edits as previous section.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1794/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='36'/>

Comment (kapil-varshney) [inline guide/14-deep-learning/use_mistral_llm_for_text_classification_and_entity_recognition.ipynb:1]: <p>Same edits as previous section.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1794/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='39'/>

Comment (kapil-varshney) [inline guide/14-deep-learning/use_mistral_llm_for_text_classification_and_entity_recognition.ipynb:1]: <p>Same edits as previous section.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1794/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='40'/>

Comment (kapil-varshney) [inline guide/14-deep-learning/use_mistral_llm_for_text_classification_and_entity_recognition.ipynb:1]: <p>To extract named entities using the ...</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1794/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='43'/>

Comment (kapil-varshney) [inline guide/14-deep-learning/use_mistral_llm_for_text_classification_and_entity_recognition.ipynb:1]: <p>Same edits as previous section.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1794/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='46'/>

Comment (kapil-varshney) [inline guide/14-deep-learning/use_mistral_llm_for_text_classification_and_entity_recognition.ipynb:1]: <p>In this guide we saw the steps to initialize and perform inference using the Mistral LLM as a backbone with the <code>TextClassifier</code> and <code>EntityRecognizer</code> models in arcgis.learn.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1794/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='52'/>

Comment (kapil-varshney) [reply guide/14-deep-learning/use_mistral_llm_for_text_classification_and_entity_recognition.ipynb:1]: <p>A <code>TextClassifier</code> model with mistral backbone can also be created without a large dataset and by using only a few examples.</p><p> </p>

**Comment by kapil-varshney** [comment]
@BP-Ent Can you please quickly review this guide? Thanks.

**Review by kapil-varshney — APPROVED** [review]
LGTM

File: guide/14-deep-learning/use_mistral_llm_for_text_classification_and_entity_recognition.ipynb [hunk 4]
```diff
@@ -0,0 +1,615 @@
...
```
Comment (BP-Ent) [inline guide/14-deep-learning/use_mistral_llm_for_text_classification_and_entity_recognition.ipynb:1]: <ul><li><br></li><li><a href="https://app.reviewnb.com/Esri/arcgis-python-api/pull/1794/#about-mistral-model" target="_blank">Introduction to Mistral model</a> </li><li>  </li><li><a href="https://app.reviewnb.com/Esri/arcgis-python-api/pull/1794/#mistral_integration_in_arcgis_learn" target="_blank">Mistral Implementation in </a><code><a href="https://app.reviewnb.com/Esri/arcgis-python-api/pull/1794/#mistral_integration_in_arcgis_learn" target="_blank">arcgis.learn</a></code> </li><li>      </li><li><a href="https://app.reviewnb.com/Esri/arcgis-python-api/pull/1794/#install_the_model_backbone" target="_blank">Install the model backbone</a></li><li><br></li><li><a href="https://app.reviewnb.com/Esri/arcgis-python-api/pull/1794/#mistral_with_textclassifier_model" target="_blank">Mistral with </a><code><a href="https://app.reviewnb.com/Esri/arcgis-python-api/pull/1794/#mistral_with_textclassifier_model" target="_blank">TextClassifier</a></code><a href="https://app.reviewnb.com/Esri/arcgis-python-api/pull/1794/#mistral_with_textclassifier_model" target="_blank"> model</a> </li><li>        </li><li><a href="https://app.reviewnb.com/Esri/arcgis-python-api/pull/1794/#import_mistral_text_classifier" target="_blank"> Import the </a><code><a href="https://app.reviewnb.com/Esri/arcgis-python-api/pull/1794/#import_mistral_text_classifier" target="_blank">TextClassifier</a></code><a href="https://app.reviewnb.com/Esri/arcgis-python-api/pull/1794/#import_mistral_text_classifier" target="_blank"> class from </a><code><a href="https://app.reviewnb.com/Esri/arcgis-python-api/pull/1794/#import_mistral_text_classifier" target="_blank">arcgis.learn.text</a></code><a href="https://app.reviewnb.com/Esri/arcgis-python-api/pull/1794/#import_mistral_text_classifier" target="_blank"> module</a></li><li>      </li><li><a href="https://app.reviewnb.com/Esri/arcgis-python-api/pull/1794/#initialize_textclassifier_model_databunch" target="_blank">Initialize the </a><code><a href="https://app.reviewnb.com/Esri/arcgis-python-api/pull/1794/#initialize_textclassifier_model_databunch" target="_blank">TextClassifier</a></code><a href="https://app.reviewnb.com/Esri/arcgis-python-api/pull/1794/#initialize_textclassifier_model_databunch" target="_blank"> model with databunch</a></li><li>  </li><li><a href="https://app.reviewnb.com/Esri/arcgis-python-api/pull/1794/#initialize_textclassifier_model_nodatabunch" target="_blank"> Initialize the </a><code><a href="https://app.reviewnb.com/Esri/arcgis-python-api/pull/1794/#initialize_textclassifier_model_nodatabunch" target="_blank">TextClassifier</a></code><a href="https://app.reviewnb.com/Esri/arcgis-python-api/pull/1794/#initialize_textclassifier_model_nodatabunch" target="_blank"> model without databunch</a></li><li>      </li><li><a href="https://app.reviewnb.com/Esri/arcgis-python-api/pull/1794/#classify_text_using_mistral_model" target="_blank">Classify text using the mistral model</a></li><li>     </li><li><a href="https://app.reviewnb.com/Esri/arcgis-python-api/pull/1794/#load_model" target="_blank">Load the model</a></li><li><br></li><li><a href="https://app.reviewnb.com/Esri/arcgis-python-api/pull/1794/#save_model" target="_blank">Save the model</a></li><li><br></li><li><a href="https://app.reviewnb.com/Esri/arcgis-python-api/pull/1794/#mistral_with_entityrecognizer_model" target="_blank">Mistral with the </a><code><a href="https://app.reviewnb.com/Esri/arcgis-python-api/pull/1794/#mistral_with_entityrecognizer_model" target="_blank">EntityRecognizer</a></code><a href="https://app.reviewnb.com/Esri/arcgis-python-api/pull/1794/#mistral_with_entityrecognizer_model" target="_blank"> model</a> </li><li><br></li><li><a href="https://app.reviewnb.com/Esri/arcgis-python-api/pull/1794/#import_mistral_entity_recognizer" target="_blank">Import the </a><code><a href="https://app.reviewnb.com/Esri/arcgis-python-api/pull/1794/#import_mistral_entity_recognizer" target="_blank">EntityRecognizer</a></code><a href="https://app.reviewnb.com/Esri/arcgis-python-api/pull/1794/#import_mistral_entity_recognizer" target="_blank"> class from the </a><code><a href="https://app.reviewnb.com/Esri/arcgis-python-api/pull/1794/#import_mistral_entity_recognizer" target="_blank">arcgis.learn.text</a></code><a href="https://app.reviewnb.com/Esri/arcgis-python-api/pull/1794/#import_mistral_entity_recognizer" target="_blank"> module</a></li><li><br></li><li><a href="https://app.reviewnb.com/Esri/arcgis-python-api/pull/1794/#initialize_entity_recognizer_model_databunch" target="_blank">Initialize the </a><code><a href="https://app.reviewnb.com/Esri/arcgis-python-api/pull/1794/#initialize_entity_recognizer_model_databunch" target="_blank">EntityRecognizer</a></code><a href="https://app.reviewnb.com/Esri/arcgis-python-api/pull/1794/#initialize_entity_recognizer_model_databunch" target="_blank"> model with databunch</a></li><li><br></li><li><a href="https://app.reviewnb.com/Esri/arcgis-python-api/pull/1794/#initialize_entity_recognizer_model_nodatabunch" target="_blank"> Initialize the </a><code><a href="https://app.reviewnb.com/Esri/arcgis-python-api/pull/1794/#initialize_entity_recognizer_model_nodatabunch" target="_blank">EntityRecognizer</a></code><a href="https://app.reviewnb.com/Esri/arcgis-python-api/pull/1794/#initialize_entity_recognizer_model_nodatabunch" target="_blank"> model without databunch</a></li><li><br></li><li><a href="https://app.reviewnb.com/Esri/arcgis-python-api/pull/1794/#extract_entities_using_mistral_model" target="_blank">Extract entities using the mistral model</a></li><li> </li><li><a href="https://app.reviewnb.com/Esri/arcgis-python-api/pull/1794/#load_model_er" target="_blank">Load the model</a></li><li><br></li><li><a href="https://app.reviewnb.com/Esri/arcgis-python-api/pull/1794/#save_model_er" target="_blank">Save the model</a></li><li><br></li><li> <a href="https://app.reviewnb.com/Esri/arcgis-python-api/pull/1794/#conclusion" target="_blank">Conclusion</a> </li><li><br></li><li> <a href="https://app.reviewnb.com/Esri/arcgis-python-api/pull/1794/#reference" target="_blank">References</a> </li></ul><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1794/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Comment (BP-Ent) [inline guide/14-deep-learning/use_mistral_llm_for_text_classification_and_entity_recognition.ipynb:1]: <p>Mistral 7B is a decoder-based language model trained using almost 7 billion parameters designed to deliver both efficiency and high performance for real-world applications. </p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1794/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

Comment (BP-Ent) [inline guide/14-deep-learning/use_mistral_llm_for_text_classification_and_entity_recognition.ipynb:1]: <h2>Mistral with the TextClassifier model </h2><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1794/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='7'/>

Comment (BP-Ent) [inline guide/14-deep-learning/use_mistral_llm_for_text_classification_and_entity_recognition.ipynb:1]: <h3>Import the TextClassifier class from the arcgis.learn.text module </h3><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1794/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='8'/>

Comment (BP-Ent) [inline guide/14-deep-learning/use_mistral_llm_for_text_classification_and_entity_recognition.ipynb:1]: <h3>Initialize the TextClassifier model with databunch</h3><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1794/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='10'/>

Comment (BP-Ent) [inline guide/14-deep-learning/use_mistral_llm_for_text_classification_and_entity_recognition.ipynb:1]: <p>Prepare databunch for the <code>TextClassifier</code> model using the <code>prepare_textdata</code> method in <code>arcgis.learn</code>.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1794/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='11'/>

Comment (BP-Ent) [inline guide/14-deep-learning/use_mistral_llm_for_text_classification_and_entity_recognition.ipynb:1]: <p>Once the data is prepared, the <code>TextClassifier</code> model object can be instantiated with the following parameters:</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1794/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='13'/>

Comment (BP-Ent) [inline guide/14-deep-learning/use_mistral_llm_for_text_classification_and_entity_recognition.ipynb:1]: <p><code>data</code>: The databunch created using the <code>prepare_textdata</code> method. </p><p><code>backbone</code>: To use mistral as the model backbone, use <em>backbone="mistral"</em>. </p><p><code>prompt</code>: Text string describing the task and its guardrails. This is an optional parameter.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1794/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='14'/>

Comment (BP-Ent) [inline guide/14-deep-learning/use_mistral_llm_for_text_classification_and_entity_recognition.ipynb:1]: <h3>Initialize the TextClassifier model without databunch </h3><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1794/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='16'/>

Comment (BP-Ent) [inline guide/14-deep-learning/use_mistral_llm_for_text_classification_and_entity_recognition.ipynb:1]: <p>A <code>TextClassifier</code> model with a mistral backbone can also be created without a large dataset using only a few examples.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1794/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='17'/>

Comment (BP-Ent) [inline guide/14-deep-learning/use_mistral_llm_for_text_classification_and_entity_recognition.ipynb:1]: <p>T</p><p>Below are the parameters to be passed into <code>TextClassifier</code>: </p><p><code>backbone</code>: To use mistral as the model backbone, use&nbsp;<em>backbone="mistral"</em>.</p><p><code>examples</code>: User defined examples to provide the mistral model, in python dictionary format:</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1794/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='18'/>

Comment (BP-Ent) [inline guide/14-deep-learning/use_mistral_llm_for_text_classification_and_entity_recognition.ipynb:1]: <h3>Classify text using the mistral model </h3><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1794/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='20'/>

Comment (BP-Ent) [inline guide/14-deep-learning/use_mistral_llm_for_text_classification_and_entity_recognition.ipynb:1]: <p>To classify text using the mistral model, use the <code>predict</code> method from the <code>TextClassifier</code> class. The input to the method will be a text string or a list of text string.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1794/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='21'/>

Comment (BP-Ent) [inline guide/14-deep-learning/use_mistral_llm_for_text_classification_and_entity_recognition.ipynb:1]: <p>To load a saved mistral model, use the <code>from_model</code> method from the <code>TextClassifier</code> class.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1794/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='24'/>

Comment (BP-Ent) [inline guide/14-deep-learning/use_mistral_llm_for_text_classification_and_entity_recognition.ipynb:1]: <p>The following method saves the model weights and creates a Deep Learning Package (.dlpk).</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1794/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='27'/>

Comment (BP-Ent) [inline guide/14-deep-learning/use_mistral_llm_for_text_classification_and_entity_recognition.ipynb:1]: <h2>Mistral with an EntityRecognizer model </h2><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1794/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='29'/>

Comment (BP-Ent) [inline guide/14-deep-learning/use_mistral_llm_for_text_classification_and_entity_recognition.ipynb:1]: <h3>Import the EntityRecognizer class from the arcgis.learn.text module </h3><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1794/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='30'/>

Comment (BP-Ent) [inline guide/14-deep-learning/use_mistral_llm_for_text_classification_and_entity_recognition.ipynb:1]: <p> </p><p><strong>Initialize the EntityRecognizer model with databunch </strong></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1794/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='32'/>

Comment (BP-Ent) [inline guide/14-deep-learning/use_mistral_llm_for_text_classification_and_entity_recognition.ipynb:1]: <p>Prepare the databunch for the <code>EntityRecognizer</code> model using the <code>prepare_textdata</code> method in <code>arcgis.learn</code>.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1794/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='33'/>

Comment (BP-Ent) [inline guide/14-deep-learning/use_mistral_llm_for_text_classification_and_entity_recognition.ipynb:1]: <p>Once the data is prepared, the <code>EntityRecognizer</code> model object can be instantiated with the following parameters:</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1794/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='35'/>

Comment (BP-Ent) [inline guide/14-deep-learning/use_mistral_llm_for_text_classification_and_entity_recognition.ipynb:1]: <p><code>data</code>: The databunch created using the <code>prepare_textdata</code> method. </p><p><code>backbone</code>: To use mistral as the model backbone, use&nbsp;<em>backbone="mistral"</em>. </p><p><code>prompt</code>: Text string describing the task and its guardrails. This is an optional parameter.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1794/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='36'/>

Comment (BP-Ent) [inline guide/14-deep-learning/use_mistral_llm_for_text_classification_and_entity_recognition.ipynb:1]: <h3>Initialize the EntityRecognizer model without a databunch </h3><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1794/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='38'/>

Comment (BP-Ent) [inline guide/14-deep-learning/use_mistral_llm_for_text_classification_and_entity_recognition.ipynb:1]: <p>An <code>EntityRecognizer</code> model with a mistral backbone can also be created without a large dataset by using only a few examples.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1794/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='39'/>

Comment (BP-Ent) [inline guide/14-deep-learning/use_mistral_llm_for_text_classification_and_entity_recognition.ipynb:1]: <p>Below are the parameters to be passed into <code>EntityRecognizer</code> : </p><p><code>backbone</code>: To use mistral as the model backbone, use&nbsp;<em>backbone="mistral"</em>.</p><p><code>examples</code>: User defined examples for the mistral model, in python list format: </p><p><code>prompt</code>: Text string describing the task and its guardrails. This is an optional parameter. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1794/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='40'/>

Comment (BP-Ent) [inline guide/14-deep-learning/use_mistral_llm_for_text_classification_and_entity_recognition.ipynb:1]: <h3>Extract entities using the mistral model </h3><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1794/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='42'/>

Comment (BP-Ent) [inline guide/14-deep-learning/use_mistral_llm_for_text_classification_and_entity_recognition.ipynb:1]: <p>To extract named entities using the mistral model, use the <code>extract_entities</code> method from the <code>EntityRecognizer</code> class. The input to the method will be a text string or a list of text strings.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1794/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='43'/>

Comment (BP-Ent) [inline guide/14-deep-learning/use_mistral_llm_for_text_classification_and_entity_recognition.ipynb:1]: <p>To load a saved mistral model, use the <code>from_model</code> method from the <code>EntityRecognizer</code> class.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1794/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='46'/>

Comment (BP-Ent) [inline guide/14-deep-learning/use_mistral_llm_for_text_classification_and_entity_recognition.ipynb:1]: <p> </p><p>In this guide we demonstrated the steps to initialize and perform inference using the Mistral LLM as a backbone with the <code>TextClassifier</code> and <code>EntityRecognizer</code> models in <code>arcgis.learn</code>.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1794/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='52'/>

**Review by BP-Ent — DISMISSED** [review]
Suggested changes made on reviewnb

**Comment by SurajBaloni** [comment]
> Suggested changes made on reviewnb

I have made the suggested changes, thanks.

**Review by priyankatuteja — APPROVED** [review]

**Comment by priyankatuteja** [comment]
PR merged on next branch

**Pull request closed without merging**